quarkchain/cluster/jsonrpc.py

```diff
@@ -1428,5 +1420,44 @@ def shutdown(self):
         return "pong"
 
     @public_methods.add
-    async def echo(self, params):
-        return params
+    async def qkc_subscribe(self, sub_type, full_shard_id, context):
+        if context is None or full_shard_id is None:
+            return None
+        self.sub_id += 1
+        response = {"jsonrpc": "2.0", "result": self.sub_id, "id": 1}  # TODO: change id
```

Comment:
per-connection id should be the same as in `message` in line 1395, and should be available in `d` from line 1400. can pass into `context` as well